libsql/src/hrana/transaction.rs

```diff
@@ -0,0 +1,64 @@
+use crate::hrana::pipeline::{ExecuteStreamReq, StreamRequest};
+use crate::hrana::proto::{Batch, BatchResult, Stmt, StmtResult};
+use crate::hrana::stream::HttpStream;
+use crate::hrana::{HttpSend, Result};
+use crate::TransactionBehavior;
+
+#[derive(Debug, Clone)]
+pub(crate) struct HttpTransaction<T>
+where
+    T: for<'a> HttpSend<'a>,
+{
+    stream: HttpStream<T>,
+}
+
+impl<T> HttpTransaction<T>
```

Comment:
Yes, but that implies sending rollback request in non-async method context, which can also fail (and possibly deadlock in some cases). IMO rollback on drop should be implemented on server side - and probably dependent on is_autocommit behaviour as well.

Reply:
Can we tokio::spawn and rollback? So its done async to the drop and that shouldn't cause any deadlocks?

Reply:
Can we be sure that tokio task we complete if ie. program ends right after transaction drop?

Reply:
I've introduced `Drop for libsql::Transaction` together with optional close callback invoked there. It's basically used to schedule rollback task on HranaTransaction. That was the best option I could think of, given that HranaTransaction may be running in context where `tokio::spawn` is not available, and `futures::block_on` basically deadlocks the tests (or single thread runtime).

I haven't figured yet how to rollback on transaction drop in Cloudflare - it has a context method that can be used to schedule async method instead of using tokio, but the access pattern to it is very invasive in the API.

Reply:
@Horusiath I found this https://github.com/cloudflare/workers-rs/blob/main/examples/tokio-postgres/src/lib.rs#L19 I wonder if we can use this function to spawn in wasm?